crates/goose/src/tracing/observation_layer.rs

```diff
@@ -104,6 +110,15 @@ impl ObservationLayer {
     pub async fn handle_span(&self, span_id: u64, span_data: SpanData) {
         let observation_id = span_data.observation_id.clone();
 
+        if let Some(session_id) = span_data
+            .metadata
+            .get("session_id")
+            .and_then(|v| v.as_str())
+        {
+            let mut spans = self.span_tracker.lock().await;
+            spans.set_session_id(Some(session_id.to_string()));
```

Comment:
with only one `goosed` for multiple sessions, how does/would this react to session changes?